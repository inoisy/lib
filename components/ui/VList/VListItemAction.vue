<script>
import Vue from 'vue';

export default Vue.extend({
    name: 'VListItemAction',

    functional: true,

    render(h, { data, children = [] }) {
        data.staticClass = data.staticClass ? `v-list-item__action ${data.staticClass}` : 'v-list-item__action';
        const filteredChild = children.filter(VNode => VNode.isComment === false && VNode.text !== ' ');
        if (filteredChild.length > 1) {
            data.staticClass += ' v-list-item__action--stack';
        }

        return h('div', data, children);
    },
});
</script>
