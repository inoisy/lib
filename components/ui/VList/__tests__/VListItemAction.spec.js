// Components
import VListItemAction from '../VListItemAction';

// Utilities
import Vue from 'vue';
import {
    mount,
} from '@vue/test-utils';
import { functionalContext } from '../../../../../test';
/* eslint-disable vue/one-component-per-file */
describe('VListItemAction.js', () => {
    let mountFunction;

    beforeEach(() => {
        mountFunction = (options = {}) => mount(VListItemAction, {
            ...options,
        });
    });

    it('should render component and match snapshot', () => {
        const wrapper = mountFunction(functionalContext());

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render component with static class and match snapshot', () => {
        const wrapper = mountFunction(functionalContext({
            staticClass: 'static-class',
        }));

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render component with many children and match snapshot', () => {
        const content1 = mount(Vue.component('Content1', {
            render: h => h('div'),
        })).vNode;
        const content2 = mount(Vue.component('Content2', {
            render: h => h('span'),
        })).vNode;
        const wrapper = mountFunction(functionalContext({}, [content1, content2]));

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render component with one children and match snapshot', () => {
        const visible = mount(Vue.component('Visible', {
            render: h => h('div') || h(),
        })).vNode;
        const notVisible = mount(Vue.component('NotVisible', {
            render: h => h() || h('span'),
        })).vNode;

        const wrapper = mountFunction(functionalContext({}, [visible, notVisible]));

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should work with v-html', () => {
        const wrapper = mountFunction({
            context: Object.assign({
                domProps: {
                    innerHTML: '<b>something</b>',
                },
                data: {},
                props: {},
            }),
        });

        expect(wrapper.html()).toMatchSnapshot();
    });
});
