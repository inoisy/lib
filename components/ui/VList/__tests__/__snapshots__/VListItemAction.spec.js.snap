// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VListItemAction.js should render component and match snapshot 1`] = `
<div class="v-list-item__action">
</div>
`;

exports[`VListItemAction.js should render component with many children and match snapshot 1`] = `
<div class="v-list-item__action">
</div>
`;

exports[`VListItemAction.js should render component with one children and match snapshot 1`] = `
<div class="v-list-item__action">
</div>
`;

exports[`VListItemAction.js should render component with static class and match snapshot 1`] = `
<div class="v-list-item__action static-class">
</div>
`;

exports[`VListItemAction.js should work with v-html 1`] = `
<div class="v-list-item__action">
  <b>
    something
  </b>
</div>
`;
